### PR TITLE
Fix algorithm in package NEVRA filter

### PR DESCRIPTION
**Pull request by j-mracek**
DNF5 detects first, whether provided argument contains epoch (find ':') then it construct package strings IDs accordingly.

CI-tests: https://github.com/rpm-software-management/ci-dnf-stack/pull/1372

Closes: https://github.com/rpm-software-management/dnf5/issues/614

### CODE REVIEW

File: libdnf5/solv/pool.cpp [hunk 1]
```diff
@@ -79,6 +79,71 @@ Pool::~Pool() {
 }
 
 
+/// Returns a temporrary object allocated by pool_alloctmpspace
```
Comment (evan-goode) [inline libdnf5/solv/pool.cpp:82]: Spelling: temporrary -> temporary, also in `pool.hpp`

File: libdnf5/rpm/package_query.cpp [hunk 2]
```diff
@@ -1943,12 +1945,24 @@ void PackageQuery::PQImpl::filter_nevra(
             filter_nevra_internal_str<cmp_lte>(pool, c_pattern, sorted_solvables, filter_result);
             break;
         case libdnf5::sack::QueryCmp::GLOB:
-            filter_glob_internal<&libdnf5::solv::RpmPool::get_nevra>(
-                pool, c_pattern, *pkg_set.p_impl, filter_result, 0);
+            found = pattern.find(':');
```
Comment (evan-goode) [inline libdnf5/rpm/package_query.cpp:1948]: Nit: I think it would be more readable to declare `found` down here instead of above the switch

**Review by evan-goode — CHANGES_REQUESTED** [review]

File: libdnf5/solv/pool.cpp (hunk 1 above)
Comment (j-mracek) [reply libdnf5/solv/pool.cpp]: Thank you

File: libdnf5/rpm/package_query.cpp (hunk 2 above)
Comment (j-mracek) [reply libdnf5/rpm/package_query.cpp]: Fixed

**Review by evan-goode — APPROVED** [review]

**Pull request merged**